micro-rdk/src/common/data_manager.rs

```diff
@@ -27,6 +27,21 @@ use thiserror::Error;
 // the smaller amount of available RAM, we've halved it
 static MAX_SENSOR_CONTENTS_SIZE: usize = 32000;
 
+/// This technically allows for a C project using micro-RDK as a library
+/// to implement a callback to be run whenever data has successfully been uploaded.
+/// The callback should be identical in signature to the weak symbol declared here
+/// Ex. for ESP-IDF project
+/// void data_upload_callback(void) {
+///     ESP_LOGI("proj_main", "hit data upload callback");
+/// }
+/// # Safety
+/// implemented callback must manage its own memory, micro-RDK will
```

Comment:
this is a bit implied :P But I would just say : the callback should be kept simple and not do any blocking or long operation